scipy/sparse/_construct.py

```diff
@@ -44,6 +52,10 @@ def spdiags(data, diags, m=None, n=None, format=None):
     format : str, optional
         Format of the result. By default (format=None) an appropriate sparse
         matrix format is returned. This choice is subject to change.
+    sparray : bool, optional
+        Whether to return a sparse 2d array (rather than a sparse matrix).
+        By default, a sparse matrix is returned. The default will change
+        to return a sparse 2d array in the future.
```

Comment:
```suggestion
        to return a sparse 2d array in the future.
        To make code future-compatible, set `sparray=True`.
```

Maybe something like this; I'm not sure about the exact wording.